cEP-0015.md

```diff
@@ -0,0 +1,212 @@
+```
+
+## Configurability
+
+corobo has a potential to be used by other organizations for similar tasks like
+onboarding and automation. Currently, it is not configurable and many plugins
+are still very coala specific. Making it more configurable will allow other
+organizations to adapt corobo to cater their needs.
+
+Making existing coala specific plugins generic for other organizations and
+letting them configure the plugins as per their needs will ensure
+configurability.
+
+Errobot provides [plugin configuration](http://errbot.io/en/latest/user_guide/plugin_development/configuration.html)
+through the built-in `!config` command which can be used by other organizations
+to configure the plugins as per their needs.
+
+List of Plugins which are coala specific and can be generalized:
+
+1. `LabHub` plugin commands are meant to work for a coala specific team names
+   maintainers, developers, newcomers and coala repositories.
+
+   Issue Link: <https://github.com/coala/corobo/issues/382>
+
+2. `explain` currently uses hardcoded coala explanations. Configuring
+   sub-directories `explain/genric` and `explain/<org name>` so that
+   other organizations can include their culture-specific explanations.
+
+3. `searchdocs` uses API_DOCS and USER_DOCS links which are constant for coala.
+    Other organizations will be able to initialize their own documentation
+    links as per their requirement.
+
+   Issue Link: <https://github.com/coala/corobo/issues/380>
+
+4. `community_stats` will be moved out from LabHub as a separate plugin since
+   it can also be used by other organizations for analytics overview.
+
+   Issue Link: <https://github.com/coala/corobo/issues/361>
+
+5. `wolfram_alpha` plugin currently uses an environment variable, it will be
+   modified to use the configuration template.
+
+   Issue Link: <https://github.com/coala/corobo/issues/383>
+
+6. `answer` plugin use `ANSWER_END` environment variable as an endpoint of the
+   answers microservice. It'll be configured so that other organizations can
+   also adapt to it.
+
+   Issue Link: <https://github.com/coala/corobo/issues/381>
+
+### Configuration Sample
+
+```py
+class LabHub(BotPlugin):
+    def get_configuration_template(self):
+        return {'GH_TEAM_NAMES': ['newcomers', 'developers', 'maintainers']}
+
+    @botcmd
+    def mycommand(self, mess, args):
+        # oh I need my TEAM !
+        gh_teams = self.config['GH_TEAM_NAMES']
+```
+
+```
+<BOT_PREFIX> plugin config LabHub {'GH_TEAM_NAMES': ['newbies', 'devs', 'admins']}
+```
+
+### Configuration of corobo
+
+Currently using the configuration templates means that the bot admin will have
+to configure the plugin every time the bot starts and since corobo is
+continuously deployed on each commit, reconfiguring again and again using the
+chat via a admin is not preferable.
+
+Configuration of corobo will be done by using docker volumes to inject file
+with configuration dict from host machine to the container so that
```

Comment:
Who creates this 'dictionary' to begin with?

Will corobo/errbot write this file?

Have you tried this with a simple plugin which uses bot plugin config?
Or does it only work with bot plugin storage?